other/Apache Tomcat Manager Path Normalization Panel.bcheck

```diff
@@ -0,0 +1,31 @@
+metadata:
+	language: v1-beta
+	name: "Apache Tomcat Manager Path Normalization Panel - Detect"
+	author: "Parimal Shaw"
+	description: "Apache Tomcat Manager Path Normalization login panel was discovered via path normalization. Normalizing a path involves modifying the string that identifies a path or file so that it confirms to a valid path on the target operating system."
+	tags: "Apache Tomcat Misconfiguration"
+
+run for each:
+    potential_path = 
+		"/..;/manager/html",
+		"/..;/..;/manager/html;/",
+		"/..;/hostmanager/html",
+		"/..;/..;/hostmanager/html;/",
+		"/bchecktest/..;/manager/html",
+		"/bchecktest/..;/hostmanager/html"
+	
+
+given host then
+    send request called check:
+		method: "GET"
+		path: {potential_path}
+
+			if {check.response.status_code} is "403" 
+			and "username=\"tomcat\" password=\"s3cret\"" in {check.response.body} 

```

Comment:
if the user has changed the password will this line work? I have a concern you are checking this response specifically.

Reply:
No if the user has changed the password the following code will not work. As ther is no regex support in bcheck file we cannot mention a any regex value of there if regex support kindly let us know.

Reply:
There is regex support in BCheck - see `{X} matches "[regex]"` in https://portswigger.net/burp/documentation/scanner/bchecks/bcheck-definition-reference

I think at this time this check is too fragile and that explicit password check should be replaced with a regex.

Reply:
Yes , sure will update that changes with regex

Reply:
thank you

Reply:
Can you confirm rest of my bchecks.

Reply:
The changes have been made ,kindly check